assets/docs/interactivity_tutorial.md

```diff
@@ -0,0 +1,441 @@
+A LocusZoom.js plot allows the user to pan and zoom to new regions. In order to update in response to user events, the plot must be able to ask for (pull) the data it needs, rather than render whatever it was given initially (push). 
+
+This is a common sticking point when adding LocusZoom to a page with other interactive widgets that share the same data: each widget makes assumptions about the shape of the data, and it can be frustrating to cede control of data requests to any one piece of the page. Trying to force-feed data from outside of the plot (push) often leads to bugs, because the external page and LocusZoom may not agree on what data is to be shown.
+
+Fortunately, there are ways for widgets to cooperate! In the pull-oriented data retrieval model used by LocusZoom, data fetches are represented as a discrete event. Thus it is possible to recognize when the plot changes, and broadcast updates to other widgets accordingly. We provide several mechanisms (such as `plot.subscribeToData`) to make this possible.
+
+### What is remembered?
+Each time the plot is re-rendered, LocusZoom.js determines what data should be shown. Client-side features like filters and matching apply after the network request is complete: thus the data is recalculated even if the user is viewing the same region.
+
+Because LocusZoom pulls in its data, very little state is preserved within the rendering layer when the plot is redrawn. Outside of a few selected mechanisms (eg display annotations), changes to data (like calculated fields) should be added in the data adapter, not in the rendering layer.
+
+> Tip: *Do not* try to force display changes by mutating `data_layer.data` directly: the changes would be overwritten on the next re-render. 
+> 
+> The best way to create a bug free page is to use mechanisms that cooperate with the data flow, rather than trying to impose a different model imperatively from the outside.
+
+*See also*:
+ 
+* Adding custom fields while fetching data: `BaseApiAdapter.annotateData`
+* Adding interactive rendering hints such as "show a label for this point": `BaseDataLayer.setElementAnnotation`
+
+## Using external commands to update the plot
+There are many times when it would be advantageous to alter the plot in response to external events. For example: the user might define aggregation tests based on a set of variants, then update the plot with the results of the calculation.
+
+The most common mechanism for this to happen is to provide the new information as parameters in global *state* : `plot.applyState({option1: value1})`. The *state* is accessible to all data adapters, and can thus be used to modify the data requested. At the end of `applyState`, the plot is automatically re-rendered to reflect the new information.
+
+This is a surprisingly common operation. For example, to change the  genetic region shown in the plot, we often write the following code:
+`plot.applyState({chr:'1', start: 100, end: 500})`
```

Comment:
Note to self: also document that state can work without arguments and will still trigger a re-fetch + re-render. (using cache if appropriate)